### PR TITLE
feat: add log collection operations

### PR DESCRIPTION
**Pull request by Baoyuantop**

### CODE REVIEW

**Review by oil-oil — COMMENTED** [review]
More comments should be added

File: log_collection.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	// Users need to specify the ControlPlane in the `opts`.
+	DeleteLogCollection(ctx context.Context, lcID ID, opts *ResourceDeleteOptions) error
+	// GetLogCollection gets an existing API7 Cloud Log Collection in the specified control plane.
+	// The given `lcID` parameter should specify the LogCollection that you want to get.
+	// Users need to specify the ControlPlane in the `opts`.
+	GetLogCollection(ctx context.Context, lcID ID, opts *ResourceGetOptions) (*LogCollection, error)
+	// ListLogCollection returns an iterator for listing LogCollection in the specified control plane with the
+	// given list conditions.
+	// Users need to specify the ControlPlane, Paging conditions in the `opts`.
+	ListLogCollection(ctx context.Context, opts *ResourceListOptions) (LogCollectionIterator, error)
```
Comment (tokers) [inline log_collection.go:71]:
```suggestion
	// ListLogCollections returns an iterator for listing Log Collections in the specified control plane with the
	// given list conditions.
	// Users need to specify the ControlPlane, Paging conditions in the `opts`.
	ListLogCollections(ctx context.Context, opts *ResourceListOptions) (LogCollectionIterator, error)
```

File: log_collection.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
```
Comment (tokers) [inline log_collection.go:73]: Also, this interface should have more comments (and the Next method).

File: log_collection_test.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+			iter := raw.(*logCollectionIterator)
+			assert.Equal(t, tc.iterator.iter.resource, iter.iter.resource, "check resource")
+			assert.Equal(t, tc.iterator.iter.path, iter.iter.path, "check path")
+			assert.Equal(t, tc.iterator.iter.paging.Page, iter.iter.paging.Page, "check page")
+			assert.Equal(t, tc.iterator.iter.paging.PageSize, iter.iter.paging.PageSize, "check page size")
+		})
+	}
+}
+
+func TestPutLogCollection(t *testing.T) {
```
Comment (tokers) [inline log_collection_test.go:225]:
```suggestion
func TestUpdateLogCollection(t *testing.T) {
```

**Review by tokers — APPROVED** [review]

**Review by oil-oil — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Pull request merged**